cylc/flow/workflow_db_mgr.py

```diff
@@ -112,27 +115,27 @@ def __init__(self, pri_d=None, pub_d=None):
             self.TABLE_ABS_OUTPUTS: []}
         self.db_updates_map = {}
 
-    def copy_pri_to_pub(self):
-        """Copy content of primary database file to public database file.
-
-        Use temporary file to ensure that we do not end up with a partial file.
-
-        """
-        temp_pub_db_file_name = None
+    def copy_pri_to_pub(self) -> None:
+        """Copy content of primary database file to public database file."""
         self.pub_dao.close()
+        # Use temporary file to ensure that we do not end up with a
+        # partial file.
+        # If an external connection is locking the old public db, it will
+        # still be connected to its inode, but should no longer affect future
+        # accesses (hopefully that process will soon recover to give up
+        # the lock).
+        temp_pub_db_fd, temp_pub_db_file_name = mkstemp(
+            prefix=self.pub_dao.DB_FILE_BASE_NAME,
+            dir=os.path.dirname(self.pub_dao.db_file_name)
+        )
```

Comment:
Previously `IOError` was caught for the creation of this file. I think the main reason this would be raised is if the user runs out of disk space (on a network filesystem this may mean they have hit quota). In this event the public DB copy would have been skipped, now it will force scheduler shutdown.

There isn't much Cylc can do in this situation, a failed DB write would result in scheduler shutdown anyway so it's most likely the workflow is going to die anyway. I'm just a little cautious changing this behaviour as I know it has been evolved to handle nasty problems in the past (there should really be comments to reflect this!).

Reply:
But it doesn't need to be inside the `try` block, does it? If it errors out during creation of the file, the file won't actually be created, so no need to `os.remove()` it? Or is that too much of an assumption?

Reply:
Yes, no need to try `os.remove()` for something that's not created. It's more that if if errors during the creation of the file that error will bubble up to the Scheduler and force shutdown.

Reply:
Previously it would re-raise the `IOError/OSError` anyway though?
https://github.com/cylc/cylc-flow/blob/0230c9da0690ead76e9747a8f2f16bf175f3d51b/cylc/flow/workflow_db_mgr.py#L123-L136

Reply:
Ah, sorry, I missed the `raise` when reading the diff, whatever the behaviour was before is good for now.